ui/page/main_page.go

```diff
@@ -511,10 +524,39 @@ func (mp *MainPage) LayoutTopBar(gtx layout.Context) layout.Dimensions {
 											})
 									}),
 									layout.Rigid(func(gtx C) D {
+										if mp.hideBalance.Changed() {
+											mp.hide = !mp.hide
+										}
 										return layout.Center.Layout(gtx, func(gtx C) D {
-											return components.LayoutBalance(gtx, mp.Load, mp.totalBalance.String())
+											if mp.hide {
+												hidenText := mp.Theme.Label(values.TextSize24, "********")
+												return hidenText.Layout(gtx)
+											} else {
+												return components.LayoutBalance(gtx, mp.Load, mp.totalBalance.String())
+											}
 										})
 									}),
+									layout.Rigid(
+										func(gtx C) D {
+											margins := layout.Inset{
+												Top:    unit.Dp(8),
+												Left:   unit.Dp(18),
+												Bottom: unit.Dp(4),
+											}
+											return margins.Layout(gtx,
+												func(gtx C) D {
+													if mp.hide {
+														mp.text = "Show Balance   "
+													} else {
+														mp.text = "Hide Balance   "
+													}
+													// hidebal:= material.Button(mp.theme.Base, mp.hideBalance, mp.text)
+													hidebal := material.Switch(mp.Theme.Base, mp.hideBalance)
```

Comment:
Use `decredmaterial.Switch`
```suggestion
													hidebal := mp.Theme.Switch()
```